test_conformance/subgroups/test_workgroup_non_uniform_arithmetic.cpp

```diff
@@ -0,0 +1,774 @@
+//
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "procs.h"
+#include "subhelpers.h"
+#include "harness/typeWrappers.h"
+#include "workgroup_common_templates.h"
+
+
+// DESCRIPTION:
+// Test for scan inclusive non uniform functions
+// Which: 0 - add, 1 - max, 2 - min, 3 - mul, 4 - and, 5 - or, 6 - xor, 7 -
+// logical and, 8 - logical or, 9 - logical xor
+template <typename Ty, int Which> struct SCIN_NU
+{
+    static void gen(Ty *x, Ty *t, cl_int *m, int ns, int nw, int ng)
+    {
+        genrand<Ty, Which>(x, t, m, ns, nw, ng);
+    }
+
+    static int chk(Ty *x, Ty *y, Ty *mx, Ty *my, cl_int *m, int ns, int nw,
+                   int ng)
+    {
+        int ii, i, j, k, n;
+        int nj = (nw + ns - 1) / ns;
```

Comment:
What's "nj"? What's it supposed to represent?

Reply:
nj - number of subgroup